src/main/java/frc/robot/commands/VisionAssistedTargetRPM.java

```diff
@@ -0,0 +1,111 @@
+package frc.robot.commands;
+
+import frc.robot.config.Config;
+
+//import frc.robot.subsystems.DriveBase;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.nettables.VisionCtrlNetTable;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+/**
+ * VisionAssistedTargetRPM command.
+ * NOTE: this command should be run after command TurnToOuterPortCommand
+ */
+public class VisionAssistedTargetRPM implements Runnable {
+  
+  //subsystem
+    
+  //todo: can be configured in config file as well
+  private final double SHOOTER_ANGLE_IN_DEGREES  = 46.88;
+  private final double TARGET_HEIGHT_IN_METERS = 2.02;
+  private final double SHOOTER_WHEEL_RADIUS_IN_CM = 5.5;
+  private final double D_CAMERA_SHOOTER_IN_METERS = 0.26;
+
+  private final double HALF_OF_GRAVITY = 4.91;
+  private final double CONVERSION_NUMBER = 3000;
+  private final double METER_PER_FOOT = 0.3048;
+  
+  // values from Vision Network Table
+  private double distanceToOuterPortInMeters;
+  
+  //adjusted value
+  private double adjustedDistanceToOutPortInMeters;
+   
+  //network table for vision control
+  private VisionCtrlNetTable visionControlNetTable;
+
+  //calculated RPM
+  double targetDistance = 0;
+  double targetRPM = 0;
+
+  /**
+   * Creates a new VisionAssistedTargetRPM Command.
+   *
+   * @param subsystem The subsystem used by this command.
+   */
+  public VisionAssistedTargetRPM() {
+
+    // Ensure the vision is running in tape mode
+    visionControlNetTable.setTapeMode();
+    
+  }
+   
+  public void run() {
+
+    //Read the network table from vision 
+    distanceToOuterPortInMeters = visionControlNetTable.distanceToOuterPort.get();
```

Comment:
visionControlNetTable needs to be created, if done elsewhere as a singleton then we need get that instance, otherwise use new